Add GraphQL API reference (#79)
vtex/mkp-app-template: docs/marketplace-integration-graphql-api.md

This GraphQL API can retrieve and change the external marketplace integration settings.

## Schema overview

### Query

```mermaid
classDiagram
  direction LR
  class Configuration {
    accountName String
    active Boolean
    affiliateId String
    salesChannel String
    email String
    allowFranchiseAccounts Boolean
  }

  class SalesChannel {
    Id ID!
    Name String
    CountryCode String
    CultureInfo String
    TimeZone String
    CurrencyCode String
  }

  class Query {
    getConfiguration Configuration
    getSalesChannel(salesChannelId String) SalesChannel
  }

  Query --> Configuration : getConfiguration
  Query --> SalesChannel : getSalesChannel

  click Query href "#query"
  click Configuration href "#configuration"
  click SalesChannel href "#saleschannel"
```

<table>
  <thead>
    <tr>
      <th align="left">Field</th>
      <th align="right">Argument</th>
      <th align="left">Type</th>
      <th align="left">Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td colspan="2" valign="top">
        <strong id="query.getconfiguration">getConfiguration</strong>
      </td>
      <td valign="top"><a href="#configuration">Configuration</a></td>

>       <td>Endpoint to retrieve the integration configuration.</td>

Reviewer: ```suggestion
      <td>Retrieves the current marketplace integration configuration, including account settings, affiliate details, and sales channel information.</td>
```

    </tr>
    <tr>
      <td colspan="2" valign="top">
        <strong id="query.getsaleschannel">getSalesChannel</strong>
      </td>
      <td valign="top"><a href="#saleschannel">SalesChannel</a></td>

>       <td>Endpoint to retrieve the sales channel properties from the sales channel ID.</td>

Reviewer: ```suggestion
      <td>Retrieves detailed properties of a specific sales channel using its unique ID.</td>
```

    </tr>
    <tr>
      <td colspan="2" align="right" valign="top">salesChannelId</td>
      <td valign="top">String</td>
      <td>Sales Channel ID.</td>
    </tr>
  </tbody>
</table>

### Mutation

```mermaid
classDiagram
  direction LR
  class Mutation {
    saveConfiguration(config ConfigurationInput) String
    createSalesChannel(salesChannelData SalesChannelInput) String
  }

  click Mutation href "#mutation"
```

<table>
  <thead>
    <tr>
      <th align="left">Field</th>
      <th align="right">Argument</th>
      <th align="left">Type</th>
      <th align="left">Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td colspan="2" valign="top">
        <strong id="mutation.saveconfiguration">saveConfiguration</strong>
      </td>
      <td valign="top">
        String
      </td>

>       <td>Endpoint to modify the integration configuration.</td>

Reviewer: ```suggestion
      <td>Updates the integration configuration with the provided settings. Returns a success message or error if the operation fails. </td>
```

    </tr>
    <tr>
      <td colspan="2" align="right" valign="top">config</td>
      <td valign="top">
        <a href="#configurationinput">ConfigurationInput</a>
      </td>
      <td>Properties of the configuration to be saved.</td>
    </tr>
    <tr>
      <td colspan="2" valign="top">
        <strong id="mutation.createsaleschannel">createSalesChannel</strong>
      </td>
      <td valign="top">
        String
      </td>

>       <td>Endpoint to create or modify a new sales channel.</td>

Reviewer: ```suggestion
      <td> Creates or updates a sales channel with the provided data. Returns a success message or error if the operation fails. </td>
```

    </tr>
    <tr>
      <td colspan="2" align="right" valign="top">salesChannelData</td>
      <td valign="top">
        <a href="#saleschannelinput">SalesChannelInput</a>
      </td>
      <td>Properties of the sales channel to be created or modified.</td>
    </tr>
  </tbody>
</table>

## Schema description

>⚠️ Arguments must be provided by the user. Required arguments are marked with an exclamation mark (`!`).

### Configuration

<table>
  <thead>
    <tr>
      <th align="left">Field</th>
      <th align="left">Type</th>
      <th align="left">Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td valign="top">
        <strong id="configuration.accountname">accountName</strong>
      </td>
      <td valign="top">
        String
      </td>
      <td>Account name where the integration is installed.</td>
    </tr>
    <tr>
      <td valign="top">
        <strong id="configuration.active">active</strong>
      </td>
      <td valign="top">
        Boolean
      </td>

>       <td>Property defining if the integration is active. If <code>true</code>, the integration is active. If <code>false</code>, the integration is not active.</td>

Reviewer: ```suggestion
      <td>Indicates whether the integration is active. If <code>true</code>, the integration is active. If <code>false</code>, the integration is not active.</td>
```

    </tr>
    <tr>
      <td valign="top">
        <strong id="configuration.affiliateid">affiliateId</strong>
      </td>
      <td valign="top">
        String
      </td>

>       <td>ID (3 consonants) to represent the connection with the external marketplace. Needs to be unique in the VTEX seller account. Example: <code>LVL</code>.</td>

Reviewer: ```suggestion
      <td>ID (3 consonants) representing the connection with the external marketplace. Needs to be unique in the VTEX seller account. Example: <code>LVL</code>.</td>
```

    </tr>
    <tr>
      <td valign="top">
        <strong id="configuration.saleschannel">salesChannel</strong>
      </td>
      <td valign="top">
        String
      </td>
      <td>Sales channel ID. It defines the catalog that will be sent to the marketplace. Example: <code>1</code>.</td>
    </tr>
    <tr>
      <td valign="top">
        <strong id="configuration.email">email</strong>
      </td>
      <td valign="top">
        String
      </td>

>       <td>Email that will receive a VTEX system notification when something happens to the affiliate.</td>

Reviewer: ```suggestion
      <td>Email that will receive VTEX notifications related to the affiliate.</td>
```

    </tr>
    <tr>
      <td valign="top">
        <strong id="configuration.allowfranchiseaccounts">allowFranchiseAccounts</strong>
      </td>
      <td valign="top">
        Boolean
      </td>

>       <td>Property defining if the integration allows franchise accounts. If <code>true</code>, the integration allows. If <code>false</code>, the integration does not allow.</td>

Reviewer: ```suggestion
      <td>Indicates whether the integration allows franchise accounts. If <code>true</code>, the integration allows. If <code>false</code>, the integration does not allow.</td>
```

    </tr>
  </tbody>
</table>

<div style="text-align: right"><a href="#query">Query</a> 🔼</div>

### ConfigurationInput

<table>
  <thead>
    <tr>
      <th align="left">Field</th>
      <th align="left">Type</th>
      <th align="left">Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td valign="top">
        <strong id="configuration.accountname">accountName</strong>
      </td>
      <td valign="top">
        String
      </td>
      <td>Account name where the integration is installed.</td>
    </tr>
    <tr>
      <td valign="top">
        <strong id="configuration.active">active</strong>
      </td>
      <td valign="top">
        Boolean
      </td>
      <td>Property defining if the integration is active. If <code>true</code>, the integration is active. If <code>false</code>, the integration is not active.</td>
    </tr>
    <tr>
      <td valign="top">
        <strong id="configuration.affiliateid">affiliateId</strong>
      </td>
      <td valign="top">
        String
      </td>

>       <td>ID (3 consonants) to represent the connection with the external marketplace. Needs to be unique in the VTEX seller account. Example: <code>LVL</code>.</td>

Reviewer: ```suggestion
      <td>ID (3 consonants) representing the connection with the external marketplace. Needs to be unique in the VTEX seller account. Example: <code>LVL</code>.</td>
```

    </tr>
    <tr>
      <td valign="top">
        <strong id="configuration.saleschannel">salesChannel</strong>
      </td>
      <td valign="top">
        String
      </td>
      <td>Sales channel ID. It defines the catalog that will be sent to the marketplace. Example: <code>1</code>.</td>
    </tr>
    <tr>
      <td valign="top">
        <strong id="configuration.email">email</strong>
      </td>
      <td valign="top">
        String
      </td>

>       <td>Email that will receive a VTEX system notification when something happens to the affiliate.</td>

Reviewer: ```suggestion
      <td>Email that will receive VTEX notifications related to the affiliate.</td>
```
